### PR TITLE
[PVR] Make shared channel selection optional.

### PR DESCRIPTION
**Pull request by ksooo**
It has been requested in the forum many times to make the "shared channel selection feature" optional. Here we go.

Runtime-tested on macOS, latest Kodi master.

@xhaggi (the biggest fan of the shared channel selection feature) mind taking a look...

### CODE REVIEW

**Comment by ksooo** [comment]
> And what bothered the users?

 They don't understand the logics and want it simple stupid. And I'm with them.

**Comment by xhaggi** [comment]
They don't understand that the channel selected in channel list will be the same in guide window  and vice versa if opened up?

**Comment by ksooo** [comment]
Yes, they want the behavior every settop box has - nothing semi cute...

**Comment by ksooo** [comment]
> They don't understand that the channel selected in channel list will be the same in guide window and vice versa if opened up?

 Maybe they understand this, but they do not want it.

**Comment by xhaggi** [comment]
Does it mean a settop box will have different selected channels in each view?

**Comment by ksooo** [comment]
Come on! What is goal of this discussion? Let's make this a setting and everybody is happy. Obviously many users don't want the current behavior and some like it. There is no right and wrong here. This is a clear indicator for a setting.

**Comment by xhaggi** [comment]
Discussions are part of pull requests and should give us the opportunity to think about the changes someone what to merge. My discussion does not mean I disagree with it in general. Please don't get me wrong. 

>What is goal of this discussion?
As you only select the playing channel in case the setting is false, and if not, nothing is changed in behavior, I'm not sure if we should name the setting like you did in this PR. Please make it clear to the user that he enable auto-select the playing channel entry in the channel list if he open the window/dialog.

File: addons/resource.language.en_gb/resources/strings.po [hunk 1]
```diff
@@ -18475,7 +18479,10 @@ msgctxt "#36213"
 msgid "Open the group manager, which allows modification of groups and their respective channels"
 msgstr ""
 
-#empty string with id 36214
+#: system/settings/settings.xml
+msgctxt "#36214"
+msgid "Share the selected channel across windows and dialogs. If enabled, the channel previously selected in a window or dialog will be pre-selected when opening any window containing a channel list. If disabled and there is a playing channel, the playing channel will be pre-selected when openeing a window or dialog containing a channel list."
```
Comment (MartijnKaijser) [inline addons/resource.language.en_gb/resources/strings.po]: Typo in "when openeing a window"

**Comment by ksooo** [comment]
That's part of the help text. I tried my best to explain this there. I like the naming of the setting exactly like it is.

**Comment by ksooo** [comment]
Nobody is complaining about the selection when nothing is playing. People complain about what is selected when something is playing. :-) 

If you want I can easily change the behavior to vreturn empty path if nothing is playing and the setting is off, which leads in selecting the first list entry.

**Comment by xhaggi** [comment]
No pls do not change the synced selection between channel lists, this should be the default if no playing channel is found.

It would be nice if we could find a better wording here. "Shared channel selection" sounds wrong to me, if that is our default if no playing channel is found. Why not the other way around, like "Select playing channel in lists", default false.

**Comment by ksooo** [comment]
Better now?
 
screenshot002

**Review by xhaggi — APPROVED** [review]
Yes thanks ;)

**Pull request merged**